containerd-shim-spin-config.md

```diff
@@ -0,0 +1,235 @@
+
+```yaml
+containers:
+- name: foo
+  env:
+  - name: NODE_IP
+    valueFrom:
+      fieldRef:
+        fieldPath: status.hostIP
+  - name: OTEL_EXPORTER_OTLP_ENDPOINT
+    value: "http://$(NODE_IP):4318"
+  - name: FOO
+    value: "bar"
+```
+
+#### Shim Execution Environment Variables Workaround
+
+Containerd shims inherit the containerd environment variables. Therefore, a shim environment variable can be configured by setting it on the containerd process. For example, to configure the `RUST_LOG` level:
+
+```sh
+RUST_LOG=info containerd --address /tmp/containerd.sock
+```
+
+If the process is managed by systemd, you can add an environment variable to the systemd service file. First, stop the service and bring up the service file:
+
+```sh
+systemctl stop containerd
+systemctl edit containerd
+```
+
+At the top of the file, add the environment variable to the service:
+
+```sh
+[Service]
+Environment="RUST_LOG=info"
+```
+
+Finally, restart the containerd service:
+
+```sh
+systemctl restart containerd
+```
+
+> note, for k3s, edit the k3s service (`systemctl edit k3s`)
+
+## Proposal
+
+As laid out in the [background](#background), this SKIP aims to enable the following categories of configuration:
+
+1. Spin execution configuration
+2. Shim execution configuration
+3. Shim execution environment variables
+
+The first category discusses how to configure an app's execution, while the later two discuss how to configure the platform.
+
+### User Interface
+
+How does a user set this configuration? As is standard with Kubernetes, configuration should be set in Custom Resource Definitions. Rather than creating a new CRD for configuration, the `SpinExecutor` and `SpinApp` CRDs can be extended to support configuring the platform and app execution, respectively. Users should be able to configure Spin execution next to their app configuration in the `SpinApp` CRD. On the other hand, platform configuration is not done on a per app basis. It could be configured in the `SpinAppExecutor` CRD. This would help users easily see how the platform is configured for the executor their app is using. It would also enable creating multiple executors with distinct configuration.
+
+### Configuring Spin Execution in Container Environment Variables
+
+All Spin execution configuration will be defined as known environment variables with an expected range of values. All trigger configuration keys have the prefix "SPIN_<TRIGGER NAME>". Each executor will define which environment variables it supports. The following is a subset of configuration values that may be supported:
+
+| Key | Spin CLI |  Example Value|
+| ---- | ---- | ---- |
+| SPIN_HTTP_LISTEN_ADDR | `spin up --listen` | "0.0.0.0:3000" |
+| OTEL_EXPORTER_OTLP_ENDPOINT | `OTEL_EXPORTER_OTLP_ENDPOINT=http://123.4.5.6:4318" spin up` | "http://123.4.5.6:4318" |
```

Comment:
Agree we are going to run into some issues with that variable and others like `RUST_LOG`.  I tried to capture the issue in runwasi  https://github.com/containerd/runwasi/issues/619

Reply:
I want to clarify that the `OTEL_EXPORTER_OTLP_ENDPOINT` is for the spin engine not the wasm instance. I do acknowledge that this could overwrite the runwasi one. For the `runwasi` env var, the user will need to set the env var in containerd. With this env var, the user could choose a different exporter for a specific app execution -- though it will miss some of the initial runwasi telemetry which will still be configured to the initial endpoint. @calebschoepp for thoughts.

Reply:
since this is for the spin engine, should we prefix it with `SPIN_` as we do for some of the other spin engine variables or is that value read by the spin engine?  Is there a way to manually set it instead of using ENV?  Will they be different endpoints in practice?

Reply:
I agree that we should add the prefix. I updated the doc. 

>  Is there a way to manually set it instead of using ENV

I am not sure i understand what you mean by manually

Reply:
I think I meant in the runtime-config.toml or spin.toml